### PR TITLE
Reject activation of unsupported elements in process instance modification

### LINKED ISSUE #9729: Reject the modification command if an element for activation is of an unsupported type
## Description

The modification command can contain activate instructions for different elements but not all elements are supported. Currently, we are not able to activate all types of elements (mainly, for technical reasons). 

If the modification command has an activate instruction for an unsupported element type then the command should be rejected.

The following element types are not supported:
- start events (i.e. root and non-root start events)
- sequence flows
- boundary events
- events that belong to an event-based gateway 

The following process shows which elements can be activated (green) and which elements are not supported (red).

c8-activate-element

Related to 
- #9408

Blocked by 
- #9639

### PR DESCRIPTION
**Pull request by korthout**
## Description

The modification command can contain activation instructions for different elements. The elements are identified by their element ids. However, some element types are unsupported for technical reasons. If at least one element id refers to an unsupported element then the command is rejected as INVALID_ARGUMENT.

The reason for the rejection contains a reference to the Bpmn Process ID and all of the ids of the unsupported elements, as well as a description of what really is unsupported. 

## Examples
For example, when referring to multiple elements that are unsupported you might see:

- `Expected to modify instance of process 'process' but it contains one or more activate instructions for elements that are unsupported: 'root_start', 'flow_from_A', 'timer_boundary_on_user_task'. The activation of elements with type 'START_EVENT', 'SEQUENCE_FLOW', 'BOUNDARY_EVENT' is not supported. Supported element types are: [BUSINESS_RULE_TASK, EVENT_BASED_GATEWAY, INTERMEDIATE_CATCH_EVENT, END_EVENT, PROCESS, USER_TASK, MANUAL_TASK, CALL_ACTIVITY, SEND_TASK, INCLUSIVE_GATEWAY, EVENT_SUB_PROCESS, INTERMEDIATE_THROW_EVENT, SUB_PROCESS, RECEIVE_TASK, EXCLUSIVE_GATEWAY, MULTI_INSTANCE_BODY, SCRIPT_TASK, SERVICE_TASK, PARALLEL_GATEWAY]`

Some special messages are set up for:
- events that belong to an event-based gateway
  - `Expected to modify instance of process 'process' but it contains one or more activate instructions for elements that are unsupported: 'gateway_timer_event'. The activation of events belonging to an event-based gateway is not supported. Supported element types are: [EVENT_SUB_PROCESS, EXCLUSIVE_GATEWAY, RECEIVE_TASK, INCLUSIVE_GATEWAY, SUB_PROCESS, USER_TASK, MANUAL_TASK, SEND_TASK, END_EVENT, PROCESS, SERVICE_TASK, EVENT_BASED_GATEWAY, PARALLEL_GATEWAY, BUSINESS_RULE_TASK, SCRIPT_TASK, INTERMEDIATE_THROW_EVENT, MULTI_INSTANCE_BODY, CALL_ACTIVITY, INTERMEDIATE_CATCH_EVENT]`
- elements that reside inside a multi-instance subprocess (this PR slightly adjusts this already existing rejection message)
  - `Expected to modify instance of process 'process' but it contains one or more activate instructions for elements that are unsupported: 'B', 'C'. The activation of elements inside a multi-instance subprocess is not supported. Supported element types are: [INTERMEDIATE_CATCH_EVENT, SCRIPT_TASK, SUB_PROCESS, PARALLEL_GATEWAY, MULTI_INSTANCE_BODY, SERVICE_TASK, SEND_TASK, BUSINESS_RULE_TASK, CALL_ACTIVITY, EVENT_BASED_GATEWAY, INCLUSIVE_GATEWAY, MANUAL_TASK, END_EVENT, PROCESS, USER_TASK, EVENT_SUB_PROCESS, INTERMEDIATE_THROW_EVENT, RECEIVE_TASK, EXCLUSIVE_GATEWAY]`

## Related issues

closes #9729 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
🚀 Nice work @korthout! I like the way you setup your test.

I have some small suggestions, please have a look at them!

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java [hunk 1]
```diff
@@ -48,11 +49,16 @@
       "Expected to modify process instance but no process instance found with key '%d'";
   private static final String ERROR_MESSAGE_TARGET_ELEMENT_NOT_FOUND =
       "Expected to modify instance of process '%s' but it contains one or more activate instructions with an element that could not be found: '%s'";
-  private static final String ERROR_MESSAGE_TARGET_ELEMENT_IN_MULTI_INSTANCE_BODY =
+  private static final String ERROR_MESSAGE_TARGET_ELEMENT_UNSUPPORTED =
       "Expected to modify instance of process '%s' but it contains one or more activate instructions"
-          + " for elements inside a multi-instance subprocess: '%s'. The activation of elements inside"
-          + " a multi-instance subprocess is not supported.";
-
+          + " for elements that are unsupported: '%s'. %s. Supported element types are: %s";
```
[the remark this answers is not in the document]
Comment (korthout) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java]: That is a really good point. Thanks 👍

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceUnsupportedElementsTest.java [hunk 2]
```diff
@@ -80,7 +80,25 @@ public static Collection<Object> scenarios() {
...
+                "The activation of events belonging to an event-based gateway is not supported")),
+        new Scenario(
+            "Activate start events",
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .eventSubProcess(
+                    "event_sub_process",
+                    e -> e.startEvent("sub_start", s -> s.timerWithDuration(Duration.ofHours(1))))
+                .startEvent("root_start")
+                .userTask("A")
+                .done(),
```
[the remark this answers is not in the document]
Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceUnsupportedElementsTest.java]: I've adjusted the test :+1

**Review by remcowesterhoud — APPROVED** [review]
🏅 

bors merge

**Pull request merged**